src/Filters/Actions/FilterAction.php

```diff
@@ -5,5 +5,4 @@
 interface FilterAction
 {
     public function parse();
-    public function __construct(array $params = []);
```

Comment:
Feels a bit awkward to have so many different constructors in these common classes that implement the same filter action interface. Why don't we keep the constructor common as an associative array, then have defaults() method or array inside each class that sets the default optional params to null values or whatever is required and then have a check for required parameters with proper exception being thrown - I liked the way it previously threw a custom filter exception - this way, calling code can decide what to do in case something is wrong instead of raising an argument error that something's missing?

Reply:
Some changes made please check

Reply:
I like that, thanks!